core/src/main/java/org/aerogear/mobile/core/MobileCore.java

```diff
@@ -2,273 +2,153 @@
-     *                         initialized in.
-     * @return a sorted list of services.
-     * @throws BootstrapException if circular or undefined dependencies are detected.
+     * @param context Application context
+     * @param options AeroGear initialization options
      */
-    private List<String> sortServicesIntoBootstrapOrder(List<String> declaredServices) {
-        List<String> workingDeclaredServicesList = new ArrayList<>(declaredServices);
-        List<String> sortedServices = new ArrayList<>(declaredServices.size());
-
-        while (!workingDeclaredServicesList.isEmpty()) {
-            sortedServices.add(popNextService(workingDeclaredServicesList, sortedServices));
+    public static void init(Context context, Options options) throws BootstrapException {
+        if (instance == null) {
+            instance = new MobileCore(context, options);
         }
+    }
 
-        return sortedServices;
+    @VisibleForTesting()
+    public static void cleanup() {
+        instance = null;
     }
 
     /**
-     * Removes a service from the list that can be instanciated or has all of its dependencies in
-     * sortedServices.
+     * Creates a MobileCore instance
      *
-     * @param workingList    a mutable list to find the first element in that can be resolved given
-     *                       the values in sortedServices
-     * @param sortedServices services which have had their dependencies check and are in initialization
-     *                       order
-     * @return the next service name
-     * @throws BootstrapException if circular or undefined dependencies are detected.
+     * @param context Application context
      */
-    private String popNextService(List<String> workingList, List<String> sortedServices) {
-        boolean unresolvableServiceDetected = true;
-        String serviceUnderInstanciation = "";
+    private MobileCore(Context context, Options options) throws BootstrapException {
+        this.configFileName = options.configFileName;
+
+        // -- Parse JSON config file
+        try (InputStream configStream = context.getAssets().open(configFileName)) {
+            this.servicesConfig = MobileCoreJsonParser.parse(configStream);
+        } catch (JSONException | IOException exception) {
+            String message = String.format("%s could not be loaded", configFileName);
+            throw new BootstrapException(message, exception);
+        }
 
-        for (String serviceName : workingList) {
-            List<String> dependencies = serviceRegistry.getDependenciesFor(serviceName);
-            serviceUnderInstanciation = serviceName;
-            if (dependencies.isEmpty() || sortedServices.containsAll(dependencies)) {
-                unresolvableServiceDetected = false;
-                break;
+        // -- Setting default http layer
+
+        if (options.getHttpServiceModule() == null) {
+            ServiceConfiguration configuration = this.servicesConfig.get("http");
```

Comment:
Configuration will not have http entry. It's there only because it was added manually. 
Suggestion is to remove this as it may be misleading.

Reply:
I left that because @secondsun are using it on his demo app. @secondsun any opinion about this?

Reply:
@danielpassos Good point. We may need to write mini demo and documentation for this again.

Reply:
So the theory was we need an http provider, we need to be able to configure it, hey let's use configuration from mobile-services.  It is probably not a great idea, but I can see configuration being passed in from the build plugins based on various configuration options.

Reply:
@secondsun - That is still valid use case. Going to document that in separate PR.